### ISSUE TITLE
msal-node 1.8.0 with a proxy server doesn't work

### ISSUE THREAD

**Issue by baelene** [AzureAD/microsoft-authentication-library-for-js#6230]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

1.18.0

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

Not Applicable

### Public or Confidential Client?

Confidential

### Description

Hello,

I cannot use msal-node 1.8.0 with a proxy server. I've also tried version 14,16,17

My package.json contains:

In my package.json

     "@azure/msal-node": "1.18.0",
     "http-proxy-agent": "^5.0.0",
     "https-proxy-agent": "^5.0.0"

And In the proxy, it looks like microsoftonline close the connection with a 302

URL: http://login.microsoftonline.com/$mytenantId/v2.0/.well-known/openid-configuration
Status: Redirect
Response Code: 302 Found
Method: GET
Content Type: -

Content-Length: 0
Location: https://login.microsoftonline.com:443/$mytenantId/v2.0/.well-known/openid-configuration
client-request-id: 81204f1b-f103-4066-8c03-c98294d230db
Date: Thu, 06 Jul 2023 11:13:49 GMT
Connection: close

If I curl the same  URL with the proxy, I receive a 200 with the correct payload.

Thank you for your help.

### Error Message

[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Verbose - Found cloud discovery metadata in the config.
ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retr authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://login.microsoftonline.com/$mytenantId/v2.0/.well-known/openid-configuration
at ClientAuthError.AuthError [as constructor] (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:499:24)
at new ClientAuthError (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:819:28)
at Function.ClientAuthError.createEndpointDiscoveryIncompleteError (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:
at Function. (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:8391:47)
at step (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:79:23)
at Object.throw (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:60:53)
at rejected (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:51:65)
at processTicksAndRejections (internal/process/task_queues.js:95:5) {
errorCode: 'endpoints_resolution_error',
errorMessage: 'Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your aell-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://login.microsoftonline.com/$mytenantId-configuration',
subError: ''

### Msal Logs

[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Info - getAuthCodeUrl called
[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Verbose - initializeRequestScopes called
[Thu, 06 Jul 2023 11:13:49 GMT] : [7ffb7816-8a7e-4a3e-8d53-93a8ad2ecbc5] : @azure/msal-node@1.18.0 : Verbose - buildOauthClientConfiguration called
[Thu, 06 Jul 2023 11:13:49 GMT] : [7ffb7816-8a7e-4a3e-8d53-93a8ad2ecbc5] : @azure/msal-node@1.18.0 : Verbose - building oauth client configuration with the authority: https://lod8c210-54c7-4039-a88b-4417f80773fe
[Thu, 06 Jul 2023 11:13:49 GMT] : [7ffb7816-8a7e-4a3e-8d53-93a8ad2ecbc5] : @azure/msal-node@1.18.0 : Verbose - createAuthority called
[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Verbose - Attempting to get cloud discovery metadata in the config
[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Verbose - The host is included in knownAuthorities. Creating new cloud discovery metadata from the host.
[Thu, 06 Jul 2023 11:13:49 GMT] : @azure/msal-node@1.18.0 : Verbose - Found cloud discovery metadata in the config.
ClientAuthError: endpoints_resolution_error: Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retr authority and verify the .well-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://login.microsoftonline.com/$mytenantId/v2.0/.well-known/openid-configuration
at ClientAuthError.AuthError [as constructor] (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:499:24)
at new ClientAuthError (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:819:28)
at Function.ClientAuthError.createEndpointDiscoveryIncompleteError (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:
at Function. (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:8391:47)
at step (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:79:23)
at Object.throw (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:60:53)
at rejected (/Users/eric/dev/ecosys/ecosys-src/ecosys-core-frontend/node_modules/@azure/msal-common/dist/index.cjs.js:51:65)
at processTicksAndRejections (internal/process/task_queues.js:95:5) {
errorCode: 'endpoints_resolution_error',
errorMessage: 'Error: could not resolve endpoints. Please check network and try again. Detail: ClientAuthError: openid_config_error: Could not retrieve endpoints. Check your aell-known/openid-configuration endpoint returns the required endpoints. Attempted to retrieve endpoints from: https://login.microsoftonline.com/$mytenantId-configuration',
subError: ''

### MSAL Configuration

```javascript
{
auth: {
clientId: module.config.azure.clientId,
authority: https://login.microsoftonline.com/${module.config.azure.tenantId},
clientSecret: module.config.azure.clientSecret,
knownAuthorities: [https://login.microsoftonline.com/${module.config.azure.tenantId}]
},
system: {
loggerOptions: {
loggerCallback(loglevel, message, containsPii) {
console.log(message);
},
piiLoggingEnabled: false,
logLevel: msal.LogLevel.Verbose,
},
proxyUrl: module.config.azure.proxyUrl
}
}
```

### Relevant Code Snippets

```javascript
const azureAdConfig = {
            auth: {
                clientId: module.config.azure.clientId,
                authority: `https://login.microsoftonline.com/${module.config.azure.tenantId}`,
                clientSecret: module.config.azure.clientSecret,
                knownAuthorities: [`https://login.microsoftonline.com/${module.config.azure.tenantId}`]
            },
            system: {
                loggerOptions: {
                    loggerCallback(loglevel, message, containsPii) {
                        console.log(message);
                    },
                    piiLoggingEnabled: false,
                    logLevel: msal.LogLevel.Verbose,
                },
                proxyUrl: module.config.azure.proxyUrl
            }
        };
        const cca = new msal.ConfidentialClientApplication(azureAdConfig);

        module.expressApplication.get('/:context/azureadauth', async (req, res) => {
            const authCodeUrlParameters = {
                scopes: ['User.Read'],
                redirectUri: `${module.config.azure.callbackURLPrefix}/${req.params.context}/azureadcallback`,
            };
            const authUrl = await cca.getAuthCodeUrl(authCodeUrlParameters);
            res.redirect(authUrl);
        });
```

### Reproduction Steps

Open browser with azureadauth url and I never received the redirect url.

### Expected Behavior

To receive the redirect url

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Source

Internal (Microsoft)

**Comment by bgavrilMS** [comment]
@Robbie-Microsoft - can you please provide guidance here? 

@baelene - can you try with the newly released 2.0.0 version?

**Comment by Robbie-Microsoft** [comment]
I agree with @bgavrilMS, we just a saw a similar issue resolved after upgrading to msal-node v2.0.1. @baelene, can you upgrade to this version and post the verbose logs here if you are still experiencing the same issue?

Additionally, I also seem to remember that msal-node had issues when being used in tandem with the http-proxy-agent(s) packages.

Lastly, @baelene, have you tried implementing your own Http Client? Instructions on how to do so can be found in [this sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-node-samples/custom-INetworkModule-and-network-tracing). There are instructions on how to perform network tracing as well.
